rosetta/validator/requests.go

```diff
@@ -88,11 +89,23 @@ func (v *Validator) Request(request interface{}) error {
 
 	switch msg {
 	case rosetta.BlockLength:
-		return rosetta.ErrBlockLength
+		return failure.InvalidBlockHash{
```

Comment:
By (ab)using the `param` argument for `ReportError` function from the validator library, I could also pass the "have length" information for the block hash/account address/transaction hash. So we could indeed have the `want` and `have` values for this.

The downside is - since the `param` is a string, I'd need to convert the value from integer to string, then parse it back, which isn't great.

`ReportError(field interface{}, fieldName, structFieldName string, tag, param string)`

Reply:
I am fine either way. It's not a huge deal to parse it both ways, and it's not a huge deal to be missing the want/have values IMO, so I'd say go with what you prefer, unless someone else has a strong opinion on this.

If I had to chose I'd prefer going with your solution, as this improves consistency when it comes to our error messages, which should be the priority in an open source project (consistent UX).